File: src-tauri/src/events.rs
//! Event names emitted to the WebView via `app.emit`. Must stay in sync with
//! `src/constants/events.ts`.

pub const CLIPBOARD_UPDATED: &str = "clipboard-updated";
pub const HISTORY_CLEARED: &str = "history-cleared";
pub const LICENSE_STATE_CHANGED: &str = "license-state-changed";
pub const UPDATE_AVAILABLE: &str = "update-available";

/// Emitted when the macOS panel resigns key (focus lost) so the frontend can
/// reset transient UI state. Must stay in sync with `src/constants/events.ts`.
pub const PANEL_DISMISSED: &str = "panel-dismissed";

File: src/constants/events.ts
// Event names received from the Rust backend via @tauri-apps/api/event listen.
// Must stay in sync with src-tauri/src/events.rs.

export const CLIPBOARD_UPDATED = "clipboard-updated";
export const HISTORY_CLEARED = "history-cleared";
export const LICENSE_STATE_CHANGED = "license-state-changed";
export const UPDATE_AVAILABLE = "update-available";

// Emitted when the macOS panel resigns key (focus lost) so the frontend can
// reset transient UI state. Must stay in sync with src-tauri/src/events.rs.
export const PANEL_DISMISSED = "panel-dismissed";
